src/pyflame.cc

```diff
@@ -108,63 +167,43 @@ int main(int argc, char **argv) {
     PtraceAttach(pid);
     Namespace ns(pid);
     const unsigned long tstate_addr = ThreadStateAddr(pid, &ns);
-    if (seconds) {
-      const std::chrono::microseconds interval{
-          static_cast<long>(sample_rate * 1000000)};
-      std::unordered_map<frames_t, size_t, FrameHash> buckets;
-      size_t idle = 0;
-      auto end =
-          std::chrono::system_clock::now() +
-          std::chrono::microseconds(static_cast<long>(seconds * 1000000));
-      for (;;) {
-        const unsigned long frame_addr = FirstFrameAddr(pid, tstate_addr);
-        if (frame_addr == 0) {
-          if (include_idle) {
-            idle++;
-          }
-        } else {
-          frames_t frames = GetStack(pid, frame_addr);
-          auto it = buckets.find(frames);
-          if (it == buckets.end()) {
-            buckets.insert(it, {frames, 1});
-          } else {
-            it->second++;
+    const std::chrono::microseconds interval{
+        static_cast<long>(sample_rate * 1000000)};
+    std::vector<FrameTS> call_stacks;
+    size_t idle = 0;
+    auto end = std::chrono::system_clock::now() +
+               std::chrono::microseconds(static_cast<long>(seconds * 1000000));
+    for (;;) {
+      const unsigned long frame_addr = FirstFrameAddr(pid, tstate_addr);
+      if (frame_addr == 0) {
+        if (include_idle) {
+          idle++;
+          /* Time stamp empty call stacks only if required. Since lots of time
+           * the process will be idle, this is a good optimization to have */
+          if (include_ts) {
+            auto now = std::chrono::system_clock::now();
```

Comment:
I thought about it, then i saw that we do processing such as GetStack(), update the vector with call_stack. I guess since we really don't care about micro second accuracy, it should be OK to move it to top of the loop